cpp/src/gandiva/llvm_generator.cc

```diff
@@ -295,6 +295,8 @@ Status LLVMGenerator::CodeGenExprValue(DexPtr value_expr, int buffer_count,
       selection_vector_type = types()->i64_type();
       break;
   }
+  ARROW_RETURN_IF((selection_vector_type == nullptr),
+                  Status::CodeGenError("Error creating selection_vector_type."));
```

Comment:
IMHO it would be clearer to move this into a `default` clause in the `switch` statement above.

Reply:
The switch case doesn't has a default. So I've no idea that why it reports compilation error. Maybe a stale compiler is used when building Windows in CI ?

Reply:
Hmm, if that's the case, then just remove this line?